src/Symfony/Bundle/TwigBundle/Loader/NativeFilesystemLoader.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\TwigBundle\Loader;
+
+use Twig\Error\LoaderError;
+use Twig\Loader\FilesystemLoader;
+
+class NativeFilesystemLoader extends FilesystemLoader
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function findTemplate($template, $throw = true)
+    {
+        try {
+            return parent::findTemplate($template, $throw);
+        } catch (LoaderError $e) {
+            if ('' === $template || '@' === $template[0] || !preg_match('/^([^:]*?)(?:Bundle)?:([^:]*+):(.+\.[^\.]+\.[^\.]+)$/', $template, $m)) {
```

Comment:
Can we use named patterns to make the code using `$m` more easily understandable?